test/conformance/conformance_diff_test.go

```diff
@@ -0,0 +1,131 @@
+package conformance
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/bom-squad/protobom/pkg/formats"
+	"github.com/bom-squad/protobom/pkg/reader"
+	"github.com/bom-squad/protobom/pkg/sbom"
+	"github.com/bom-squad/protobom/pkg/writer"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+)
+
+type fakeBuf struct {
+	*bufio.ReadWriter
+}
+
+func (fwc *fakeBuf) Close() error {
+	return nil
+}
+
+func (f *fakeBuf) Seek(offset int64, whence int) (int64, error) {
+	return 0, nil
+}
+
+func TestDiffTranslationFormats(t *testing.T) {
+	from := formats.SPDX23JSON
+	to := formats.CDX14JSON
+
+	fromFiles := findFiles(t, from)
+	require.NotEqual(t, len(fromFiles), 0)
+
+	fname := fromFiles[0]
+
+	r := reader.New()
+	fromSut, err := r.ParseFile(fname)
+
+	require.NoError(t, err)
+	w := writer.New()
+
+	var b bytes.Buffer
+	fakeWriter := bufio.NewWriter(&b)
+	fakeReader := bufio.NewReader(&b)
+	fakeReadWriter := bufio.NewReadWriter(fakeReader, fakeWriter)
+
+	fake := fakeBuf{
+		ReadWriter: fakeReadWriter,
+	}
+
+	w.WriteStreamWithOptions(
+		fromSut, &fake, &writer.Options{Format: to},
+	)
+	toSut, err := r.ParseStreamWithOptions(&fake, &reader.Options{Format: to})
+	require.NoError(t, err)
+
+	golden := readProtobom(t, fname+".proto")
```

Comment:
we can use fromSut but for the sake for conformance i chose to use golden proto.